changelog/0076-upgrade-to-es-5.yml

```diff
@@ -0,0 +1,279 @@
+title: Elasticsearch upgrade from 2.4.6 to 5.6.16
+key: upgrade-to-es-5
+date: 2024-01-14
+optional_per_env: no
+min_commcare_version: 3208896a555c69be119bbd7f0d06de90120b6d14
+max_commcare_version:
+context: |
+  Upgrade to Elasticsearch 5.
+
+  CommCare HQ releases after Mar 1, 2024 will not support Elasticsearch 2.x.
+  So we strongly recommend applying this change before then.
+
+details: |
+  As part of our ongoing effort to keep CommCare HQ up to date with the latest tools and
+  libraries we have upgraded Elasticsearch from version 2.4.6 to 5.6.16.
+
+update_steps: |
+  This process should be followed after Reindexing is complete for all indices. Refer to the previous [changelog](https://commcare-cloud.readthedocs.io/en/latest/changelog/0075-reindex-all-indexes-for-es-upgrade.html) for further details.
+
+  After you ensure that you have followed the steps mentioned in previous changelog, you are all set to move with upgrading your Elasticsearch cluster to version 5.
+
+  ### Pre-Upgrade Steps
+
+  1. Start by ssh-ing into the control machine.
+
+    ```
+      cchq <env> ssh control
+    ```
+
+  2. Start a new tmux session
+
+    ```
+      tmux new -s es_upgrade
+    ```
+
+  3. For commands including `${ENV}` and `${ES_HOST}` below to work, make sure to set `ENV` to the name of your environment, and `ES_HOST` to the IP address of one of your ES nodes:
+    
+    ```
+      ENV=<env>
+      ES_HOST=$(cchq ${ENV} lookup elasticsearch:0 | cut -d' ' -f1)
+    ```
+
+  4. Delete old and unused CommCare Indices by running the following commands. The command will ask you confirm the deletion by typing in the index cname. 
+    ```
+      cchq ${ENV} django-manage elastic_sync_multiplexed delete apps
+      cchq ${ENV} django-manage elastic_sync_multiplexed delete cases
+      cchq ${ENV} django-manage elastic_sync_multiplexed delete case_search
+      cchq ${ENV} django-manage elastic_sync_multiplexed delete domains
+      cchq ${ENV} django-manage elastic_sync_multiplexed delete forms
+      cchq ${ENV} django-manage elastic_sync_multiplexed delete groups
+      cchq ${ENV} django-manage elastic_sync_multiplexed delete sms
+      cchq ${ENV} django-manage elastic_sync_multiplexed delete users
+    ``` 
+  
+  5. Delete residual indices from elasticsearch that do not belong to CommCare. The command will ask you to confirm the deletion by typing in the index name.
+    
+    ```
+      cchq ${ENV} django-manage elastic_sync_multiplexed remove_residual_indices
+    ```
+
+  6. Make sure that the disk usage is well below 50%,
+  or that you have a place to back up to on another disk or remotely. You can check disk usage on each elasticsearch host usage by running:
+
+    ```
+      cchq ${ENV} run-shell-command elasticsearch  "ls -lh  /opt/data/elasticsearch-2.4.6" -b
```

Comment:
I thought `ls -lh` only gives individual file sizes. Would it make sense to use a command like `du -sh` instead?

Reply:
Yeah you are correct. This will generally list out two folders data and logs. So that's why I have not used `du`.